src/vellum/workflows/state/context.py

```diff
@@ -180,6 +180,7 @@ def resolve_workflow_deployment(
                 deployment_name,
                 release_tag=release_tag,
                 version=version_range,
+                request_options={"additional_headers": {"X-Vellum-Always-Success": "true"}},
             )
 
             zip_bytes = b"".join(response)
```

Comment:
when we pass in the above header, this response is now a valid response instead of raising an error. we should verify that `response` is not a dict before the zip logic